source/JsMaterialX/JsMaterialXView/src/index.js

```diff
@@ -7,142 +12,36 @@ import { RenderPass } from 'three/examples/jsm/postprocessing/RenderPass.js';
-        const x2 = w2.x - w0.x;
-        const y1 = w1.y - w0.y;
-        const y2 = w2.y - w0.y;
-
-        const denom = x1 * y2 - x2 * y1;
-        const r = denom ? (1.0 / denom) : 0.0;
-
-        tangent.subVectors(e1.clone().multiplyScalar(y2), e2.clone().multiplyScalar(y1)).multiplyScalar(r);
-
-        // Gram-Schmidt process
-        t0.subVectors(tangent, n0.multiplyScalar(n0.dot(tangent))).normalize();
-        t1.subVectors(tangent, n1.multiplyScalar(n1.dot(tangent))).normalize();
-        t2.subVectors(tangent, n2.multiplyScalar(n2.dot(tangent))).normalize();
-
-        tangentsdata[idx] = t0.x;
-        tangentsdata[idx + 1] = t0.y;
-        tangentsdata[idx + 2] = t0.z;
-        tangentsdata[idx + 3] = t1.x;
-        tangentsdata[idx + 4] = t1.y;
-        tangentsdata[idx + 5] = t1.z;
-        tangentsdata[idx + 6] = t2.x;
-        tangentsdata[idx + 7] = t2.y;
-        tangentsdata[idx + 8] = t2.z;
-    }
-
-    geometry.setAttribute('tangent', new THREE.BufferAttribute( tangentsdata, 3));
-}
+const materialFilename = new URLSearchParams(document.location.search).get("file");
 
-/**
- * Create a new (half)float texture containing an alpha channel with a value of 1 from a RGB (half)float texture.
- * @param {THREE.Texture} texture 
- */
-function RGBToRGBA_Float(texture) {
-    const rgbData = texture.image.data;
-    const length = (rgbData.length / 3) * 4;
-    let rgbaData;
-
-    switch (texture.type) {
-        case THREE.FloatType:
-            rgbaData = new Float32Array(length);
-            break;
-        case THREE.HalfFloatType:
-            rgbaData = new Uint16Array(length);
-            break;
-        default:
-          break;
-    }
-
-    if (rgbaData) {
-        for (let i = 0; i < length / 4; i++) {
-            rgbaData[(i * 4) + 0] = rgbData[(i * 3) + 0];
-            rgbaData[(i * 4) + 1] = rgbData[(i * 3) + 1];
-            rgbaData[(i * 4) + 2] = rgbData[(i * 3) + 2];
-            rgbaData[(i * 4) + 3] = 1.0;
-        }
-        return new THREE.DataTexture(rgbaData, texture.image.width, texture.image.height, THREE.RGBAFormat, texture.type);
-    }
+init();
 
-    return texture;
-}
 
-function prepareEnvTexture(texture, capabilities) {
-    const rgbaTexture = RGBToRGBA_Float(texture);
-    // RGBELoader sets flipY to true by default
-    rgbaTexture.flipY = false;
-    rgbaTexture.wrapS = THREE.RepeatWrapping;
-    rgbaTexture.anisotropy = capabilities.getMaxAnisotropy();
-    rgbaTexture.minFilter = THREE.LinearMipmapLinearFilter;
-    rgbaTexture.magFilter = THREE.LinearFilter;
-    rgbaTexture.generateMipmaps = true;
-    rgbaTexture.needsUpdate = true;
-
-    return rgbaTexture;
+// If no material file is selected, we programmatically create a jade material as a fallback
+function fallbackMaterial(mx, doc) {
+    const ssName = 'SR_jade';
+    const ssNode = doc.addChildOfCategory('standard_surface', ssName);
+    ssNode.setType('surfaceshader');
+    ssNode.addInput('base').setValueFloat(0.5);
```

Comment:
It is not necessary to use `addInput` here. As it is a 'standard_surface' node we can directly set the available inputs, e.g. `ssNode.setInputValueFloat('base', 0.5);`